nomad/state/state_store.go

```diff
@@ -705,24 +705,30 @@ func (s *StateStore) UpsertNode(index uint64, node *structs.Node) error {
 	return nil
 }
 
-// DeleteNode is used to deregister a node
-func (s *StateStore) DeleteNode(index uint64, nodeID string) error {
+// DeleteNode deregisters a batch of nodes
+func (s *StateStore) DeleteNode(index uint64, nodes []string) error {
+	if len(nodes) == 0 {
+		return nil
```

Comment:
Should this potentially return an error? - It probably signifies a programmer error, as an empty node-id should previously have returned a node lookup failed error.